### PR TITLE
Quantity factory to use dsl Float type as default

### PR DESCRIPTION
**Pull request by fmalatino**
**Description**
Changes to `QuantityFactory` methods that previosuly defaulted to `numpy.float64` type to use `Float` dsl type to enable specification of 32-bit or 64-bit precision.

**How Has This Been Tested?**
Tested using currently implemented unit tests
Use in builds of `pyFV3`, `pySHiELD`, and `pace`

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by bensonr — DISMISSED** [review]

File: ndsl/quantity.py [hunk 1]
```diff
@@ -298,7 +300,7 @@ def __init__(
         """
         # ToDo: [Florian 01/23] Kill the abomination.
         # See https://github.com/NOAA-GFDL/pace/issues/3
-        from ndsl.dsl.typing import Float
+        # from ndsl.dsl.typing import Float
```
Comment (oelbert) [inline ndsl/quantity.py:303]: I'd rather delete this instead of commenting it out

**Review by oelbert — CHANGES_REQUESTED** [review]
Small change, then good to go

**Review by xyuan — DISMISSED** [review]

File: ndsl/quantity.py (hunk 1 above)
Comment (fmalatino) [reply ndsl/quantity.py:303]: I have tested these changes in the following ways:

- Test branch in pyFV3
    - python=3.8 
        - translate numpy passes
        - translate dace:cpu passes
    - python=3.11
        - translate numpy passes
        - translate dace:cpu passes
- Test branch in pySHiELD
    - python=3.8
        - translate numpy passes
        - translate dace:cpu passes
    - python=3.11
        - translate numpy passes
        - translate dace:cpu passes
- Test branch in pace
    - python=3.8
        - tests/main passed
    - python=3.11
        - tests/main passed

File: ndsl/quantity.py [hunk 2]
```diff
@@ -298,7 +300,6 @@ def __init__(
         """
         # ToDo: [Florian 01/23] Kill the abomination.
         # See https://github.com/NOAA-GFDL/pace/issues/3
```
Comment (FlorianDeconinck) [inline ndsl/quantity.py:302]: Remove this before merging

**Review by FlorianDeconinck — APPROVED** [review]
Clean up the outdated comment and it's good for me!

**Review by bensonr — APPROVED** [review]

**Review by xyuan — APPROVED** [review]

**Review by oelbert — APPROVED** [review]
👍

**Review by mlee03 — APPROVED** [review]

**Pull request merged**